### PR TITLE
Expand dsmr_reader documentation

### PR DESCRIPTION
**Pull request by Glodenox**
## Proposed change
Add more information on how the sensors in DSMR Reader are best used within the Energy Dashboard. Quite a few people were having issues with this (mostly due to the changes in the state class of some entities).

Additionally, I've better clarified the difference with the DSMR integration. I'm not sure why I previously stated that it provides an instance of DSMR Reader directly, as it has nothing to do with DSMR Reader at all.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
No other information

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/dsmr_reader.markdown [hunk 1]
```diff
@@ -33,6 +33,16 @@ To use this DSMR Reader sensor integration, you need to have a DSMR Reader insta
 
 {% include integrations/config_flow.md %}
 
+## Configuring the Energy Dashboard
+
+It is most advisable to use the `sensor.dsmr_reading_*` sensors in the energy dashboard over the "merged" and "day" sources. The "reading" sensors provide the most stable source of data for Home Assistant to use. These MQTT values are part of the "Telegram: Split topic" MQTT values within DSMR Reader, so make sure to enable it.
```
Comment (frenck) [inline source/_integrations/dsmr_reader.markdown]: Why does that matter? (reading, merged, day) Every source that meets the requirements is viable for use, the result should make any difference?

File: source/_integrations/dsmr_reader.markdown [hunk 2]
```diff
@@ -33,6 +33,16 @@ To use this DSMR Reader sensor integration, you need to have a DSMR Reader insta
...
+
+| Section          | Sensor entities to configure                                                   |
+| ---------------- | ------------------------------------------------------------------------------ |
+| Grid consumption | `dsmr_reading_electricity_delivered_1`, `dsmr_reading_electricity_delivered_2` |
+| Return to grid   | `dsmr_reading_electricity_returned_1`, `dsmr_reading_electricity_returned_2`   |
+| Gas consumption  | `dsmr_consumption_gas_delivered`                                               |
```
Comment (frenck) [inline source/_integrations/dsmr_reader.markdown]: Don't use entity IDs (object IDs in this case), those are not guaranteed to be assigned.

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_integrations/dsmr_reader.markdown (hunk 1 above)
Comment (Glodenox) [reply source/_integrations/dsmr_reader.markdown]: The reading values contain a running total, while the day values get reset each day at midnight. In some cases, there might be some data loss at that time. This is especially true when the Home Assistant instance was down for a day or more.

Slightly tangential: after your change to the state classes, the other ones actually aren't viable sources any more. I still need to revert that change, as they recently fixed the bug in DSMR Reader.

File: source/_integrations/dsmr_reader.markdown (hunk 2 above)
Comment (Glodenox) [reply source/_integrations/dsmr_reader.markdown]: Do you mean to say I should use `sensor.dsmr_reading_electricity_delivered_1` and so on instead? The code blocks overflowed the fixed table layout in that case, as it wasn't possible to break at any word. I'm all ears for a solution to that, as I wasted quite a bit of time on finding some working solution for that. The script will always assign these sensor entities with these IDs and those texts are also part of the unique IDs.

If you prefer to use the names of the entities instead: they aren't unique within the integration. There's 4 of each of them. We wouldn't be helping people by using those names.

Comment (frenck) [reply source/_integrations/dsmr_reader.markdown]: > Do you mean to say I should use sensor.dsmr_reading_electricity_delivered_1 and so on instead?

No, I'm saying not to use entity IDs or object IDs at all. Just describe them instead.

> There's 4 of each of them. We wouldn't be helping people by using those names.

I suggest we improve their names in that case (if that is really problematic, is should be addressed to being with). Object and entity IDs are not guaranteed to be assigned.

Comment (Glodenox) [reply source/_integrations/dsmr_reader.markdown]: In that case I'll just leave out the table for now, as I can't describe something without using the one thing that actually describes them. I'm looking at maybe translating those default names, so I'll combine those two changes in the integration.

Comment (Glodenox) [reply source/_integrations/dsmr_reader.markdown]: A renaming of the sensors is planned, so the names will be unique. As such, I added the table again with the regular sensor names.

**Review by frenck — APPROVED** [review]
Thanks, @Glodenox 👍 

../Frenck

**Pull request merged**